Objects/call.c

```diff
@@ -364,60 +356,6 @@ _PyFunction_Vectorcall(PyObject *func, PyObject* const* stack,
 }
 
 
-/* --- PyCFunction call functions --------------------------------- */
-
-static PyObject *
-cfunction_call_varargs(PyObject *func, PyObject *args, PyObject *kwargs)
-{
-    assert(!PyErr_Occurred());
-    assert(kwargs == NULL || PyDict_Check(kwargs));
-
-    PyCFunction meth = PyCFunction_GET_FUNCTION(func);
-    PyObject *self = PyCFunction_GET_SELF(func);
-    PyObject *result;
-
-    assert(PyCFunction_GET_FLAGS(func) & METH_VARARGS);
-    if (PyCFunction_GET_FLAGS(func) & METH_KEYWORDS) {
-        if (Py_EnterRecursiveCall(" while calling a Python object")) {
```

Comment:
The recursion check is gone now. Was that intentional? Any reason why you think we don't need it any more?
Since you're dealing with a public C-API function here, I would advise to keep the current semantics.

Reply:
The recursion check is gone from `PyCFunction_Call` but it is in `PyObject_Call`.

Note that `PyCFunction_Call` is not a documented C API function, so I assumed that this minor change in behaviour (no longer doing the recursion check) would be acceptable.